lib/Traits/GeoJsonTraits.ts

```diff
@@ -1,17 +1,46 @@
+import anyTrait from "./anyTrait";
 import CatalogMemberTraits from "./CatalogMemberTraits";
+import DiscretelyTimeVaryingTraits from "./DiscretelyTimeVaryingTraits";
 import FeatureInfoTraits from "./FeatureInfoTraits";
 import MappableTraits from "./MappableTraits";
 import mixTraits from "./mixTraits";
+import ModelTraits from "./ModelTraits";
+import objectArrayTrait from "./objectArrayTrait";
 import objectTrait from "./objectTrait";
 import primitiveTrait from "./primitiveTrait";
 import StyleTraits from "./StyleTraits";
 import UrlTraits from "./UrlTraits";
 
+export class PerPropertyGeoJsonStyleTraits extends ModelTraits {
+  @anyTrait({
+    name: "Properties",
+    description:
+      "If the properties of a feature match these properties, then apply the style to that feature"
+  })
+  properties?: any;
```

Comment:
This should be `JsonObject` - see https://github.com/TerriaJS/terriajs/pull/5517#discussion_r643606957